x/subspaces/types/msgs.go

```diff
@@ -60,6 +69,76 @@ func (msg MsgCreateSubspace) MarshalJSON() ([]byte, error) {
 	return json.Marshal(temp(msg))
 }
 
+// NewMsgEditSubspace is a constructor function for MsgEditSubspace
+func NewMsgEditSubspace(subspaceID, owner, name, description, logo, editor string, subspaceType SubspaceType) *MsgEditSubspace {
+	return &MsgEditSubspace{
+		ID:           subspaceID,
+		Owner:        owner,
+		Name:         name,
+		Description:  description,
+		Logo:         logo,
+		Editor:       editor,
+		SubspaceType: subspaceType,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgEditSubspace) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgEditSubspace) Type() string { return ActionEditSubspace }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgEditSubspace) ValidateBasic() error {
+	if msg.Editor == msg.Owner {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "the owner address is equal to the editor address")
+	}
+
+	if strings.TrimSpace(msg.Name) == "" {
+		return sdkerrors.Wrap(ErrInvalidSubspaceName, "subspace name cannot be empty or blank")
+	}
+
+	_, err := sdk.AccAddressFromBech32(msg.Editor)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid editor address")
+	}
+
+	if strings.TrimSpace(msg.Owner) != "" {
+		_, err = sdk.AccAddressFromBech32(msg.Owner)
+		if err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid owner address")
+		}
+	}
+
+	if !IsValidSubspace(msg.ID) {
+		return sdkerrors.Wrap(ErrInvalidSubspaceID, "subspace id must be a valid SHA-256 hash")
+	}
+
+	if msg.Logo != DoNotModify && !commons.IsURIValid(msg.Logo) {
```

Comment:
I think this can be changed to be 

```suggestion
	if msg.Logo != DoNotModify && strings.TrimSpace(msg.Logo) != "" && !commons.IsURIValid(msg.Logo) {
```

This way we allow editors to remove the image if they want to do so